### PR TITLE
Add test coverage check to CI.yaml

### PR DESCRIPTION
**Pull request by thepetk**
## What does this PR do?

Adds a new job for coverage inside the `CI.yml` file. More detailed we are updating the command:
```yaml
code-coverage-report:
  runs-on: ubuntu-20.04
  steps:
    - name: Checkout
      uses: actions/checkout@v2.3.1
      with:
        persist-credentials: false
    - name: Set up Go 1.x
      uses: actions/setup-go@v2
      with:
        go-version: 1.19
    - name: Run tests
      run: go test -coverprofile cover.out -v ./...
    - name: Codecov
      uses: codecov/codecov-action@v2.1.0
```
This may runs twice the tests but it explicitly has an additional check and remains consistent with the `devfile/library` flow.

All preferences for codecov have been set with the addition of `.codecov.yaml`

### Which issue(s) does this PR fix

fixes https://github.com/devfile/api/issues/1164

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [x] Unit/Functional tests

  

- [x] Documentation

   

### How to test changes / Special notes to the reviewer

I have created some tests runs here: https://github.com/thepetk/devfile-alizer/actions/runs/5543302687/jobs/10119130147?pr=2

### CODE REVIEW

**Comment by thepetk** [comment]
> I noticed that in some of the other repos, we use the codecov gh action: https://github.com/devfile/library/blob/main/.github/workflows/codecov.yml#L20-L21
> 
> What do you think about:
> 
> 1. Splitting the test into a separate workflow yaml (this does mean running the tests twice)
> 2. Using the code-cov-action which would add more details to the checks section
> 
> code-cov

Yes I thnk is a good catch. I like it and is more consistent as we are using the same way with the library :)

**Comment by thepetk** [comment]
@mike-hoang updated the PR :)

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -46,4 +46,20 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2.3.1
+        with:
+          persist-credentials: false
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.19
+      - name: Run tests
+        run: go test -coverprofile cover.out -v ./...
```
Comment (mike-hoang) [inline .github/workflows/CI.yml:63]: small nit but can we change this back to `make test`

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -46,4 +46,20 @@ jobs:
...
+      - name: Codecov
+        uses: codecov/codecov-action@v2.1.0
```
Comment (mike-hoang) [inline .github/workflows/CI.yml:65]: we should update this to newer version

**Review by mike-hoang — COMMENTED** [review]
I like it! Small changes, but let's use the newest version of the codecov action to fix the annotation warnings

https://github.com/devfile/alizer/actions/runs/5543299324

**Comment by thepetk** [comment]
@mike-hoang I've bumped up both codecov and checkout :)

**Pull request merged**